packages/fxa-payments-server/src/lib/stripe.test.ts

```diff
@@ -2,10 +2,101 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { localeToStripeLocale } from './stripe';
+jest.mock('./account', () => ({
+  handlePasswordlessSignUp: jest.fn(),
+}));
+jest.mock('./apiClient');
+
+import * as apiClient from './apiClient';
+import { FXA_SIGNUP_ERROR, handlePasswordlessSignUp } from './account';
+import { handlePasswordlessSubscription, localeToStripeLocale } from './stripe';
+import { NEW_CUSTOMER, PLAN } from './mock-data';
+
+const stripeOverride = {
+  createPaymentMethod: jest.fn().mockResolvedValue({ paymentMethod: {} }),
+  confirmCardPayment: jest.fn().mockResolvedValue({}),
+};
+const apiClientOverrides = {
+  apiCreateCustomer: jest.fn().mockResolvedValue(NEW_CUSTOMER),
+  apiCreateSubscriptionWithPaymentMethod: jest.fn().mockResolvedValue({
+    latest_invoice: {
+      payment_intent: {
+        status: 'succeeded',
+      },
+    },
+  }),
+  apiRetryInvoice: jest.fn().mockResolvedValue({
+    payment_intent: {
+      status: 'succeeded',
+    },
+  }),
+  apiDetachFailedPaymentMethod: jest.fn().mockResolvedValue({}),
+};
 
 // handleSubscriptionPayment and handlePaymentIntent are tested as part of SubscriptionCreate
 
+// At the time of writing, there is no UI component that uses
+// handlePasswordlessSubscription; once we have that implementation, we should
+// test handlePasswordlessSubscription in the same fashion that we test
+// handleSubscriptionPayment.
+describe('handlePasswordlessSubscription', () => {
+  const email = 'testo@example.com';
+  const clientId = 'thebestprogram';
+  const noop = () => {};
+
+  beforeEach(() => {
+    (handlePasswordlessSignUp as jest.Mock).mockClear();
+  });
+
+  it('calls handlePasswordlessSignUp then handleSubscriptionPayment on success', async () => {
+    const onFailure = jest.fn();
+
+    (handlePasswordlessSignUp as jest.Mock).mockResolvedValue({});
+    await handlePasswordlessSubscription({
+      email,
+      clientId,
+      stripe: stripeOverride,
+      name: 'BMO',
+      card: null,
+      idempotencyKey: 'dontrepeat',
+      selectedPlan: PLAN,
+      customer: null,
+      retryStatus: undefined,
+      ...apiClientOverrides,
+      onFailure,
+      onRetry: noop,
+      onSuccess: noop,
+    });
+
+    expect(handlePasswordlessSignUp).toBeCalledWith({ email, clientId });
```

Comment:
Wouldn't we also expect `handleSubscriptionPayment` to be called (with)?

Reply:
Sure but we cannot spy on the `handleSubscriptonPayment` function that's enclosed in `handlePasswordlessSignUp`.